### PR TITLE
Enhancement/8974 Dismiss Notifications Infrastructure

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #8974 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by zutigrm — CHANGES_REQUESTED** [review]
Thanks @jimmymadon, looks good overall, I left you a few suggestions.

Also for the QAB, doesn't seem necessary to go through QA:Eng, for two console log calls, it can be either done as MR, or also as part of regular QA, since QA can also test the console snippets, they have covered such tests in the past

File: assets/js/googlesitekit/notifications/datastore/notifications.js [hunk 1]
```diff
@@ -99,6 +101,29 @@ export const actions = {
 			type: REGISTER_NOTIFICATION,
 		};
 	},
+	/**
+	 * Dismisses the given notification by its id.
+	 *
+	 * Currently, this action simply dispatches the call to the dismissed items API.
+	 * We can potentially add more notification-specific dismissal logic here in the future.
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/datastore/notifications.js:108]: This comment was included to justify the addition of a wrapper function which is not something we usually do.

File: assets/js/googlesitekit/notifications/datastore/notifications.js [hunk 2]
```diff
@@ -126,7 +151,25 @@ export const reducer = createReducer( ( state, { type, payload } ) => {
 
 export const resolvers = {};
 
-export const selectors = {};
+export const selectors = {
+	/**
+	 * Determines whether a notification is dismissed or not.
+	 *
+	 * Currently, this selector simply forwards the call to the dismissed items API.
+	 * We can potentially add more notification-specific logic here in the future.
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/datastore/notifications.js:159]: Same as above - it just makes it clearer why we want to duplicate and create a wrapper function here.

File: assets/js/googlesitekit/notifications/datastore/notifications.test.js [hunk 3]
```diff
@@ -147,5 +156,69 @@ describe( 'core/notifications Notifications', () => {
...
+				} );
+
+				const isNotificationDismissed = registry
+					.select( CORE_NOTIFICATIONS )
+					.isNotificationDismissed( 'foo' );
+				expect( isNotificationDismissed ).toBe( true );
+
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+			} );
+		} );
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/datastore/notifications.test.js:221]: Thanks - have refactored and added more tests to cover invalid and default params.

**Comment by jimmymadon** [comment]
> Also for the QAB, doesn't seem necessary to go through QA:Eng, for two console log calls, it can be either done as MR, or also as part of regular QA, since QA can also test the console snippets, they have covered such tests in the past

@zutigrm Yeah - I prefer it being QA:Eng as an engineer might be able to do a bit more testing than just using the two console scripts, say, using the Redux toolkit. And yes, it the Merge Reviewer decides, they can QA it themselves like Matt has in the past and move the issue straight to Approval from MR. But I don't want to remove the label just in case that does not happen.

**Review by zutigrm — APPROVED** [review]
Thanks @jimmymadon LGTM

> Yeah - I prefer it being QA:Eng as an engineer might be able to do a bit more testing than just using the two console scripts, say, using the Redux toolkit. And yes, if the Merge Reviewer decides, they can QA it themselves like Matt has in the past and move the issue straight to Approval from MR. But I don't want to remove the label just in case that does not happen.

Got it, thanks

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**